provider/pkg/resources/suppress_diff.go

```diff
@@ -154,36 +164,314 @@ func getTagKey(tag resource.PropertyValue) string {
-// fileSystemProtection.replicationOverwriteProtection when AWS has transitioned
-// it to REPLICATING (which happens when the filesystem becomes a replication destination).
+// suppressPropertyTransformDiffs uses CloudFormation propertyTransform specifications
+// to suppress diffs where values are semantically equivalent after transformation.
 //
-// When an EFS filesystem is used as a replication destination, AWS automatically
-// transitions replicationOverwriteProtection from DISABLED to REPLICATING. This is
-// an AWS-managed state that the user cannot change back while replication is active.
-func suppressEFSReplicationProtectionTransition(diff *resource.ObjectDiff) *resource.ObjectDiff {
-	fspKey := resource.PropertyKey("fileSystemProtection")
-
-	updated, isUpdate := diff.Updates[fspKey]
-	if !isUpdate || !updated.New.IsObject() {
+// PropertyTransforms define JSONata expressions that normalize property values for
+// comparison. For example, EC2 SecurityGroup transforms numeric IP protocols (6, 17)
+// to their string equivalents (tcp, udp), and RDS transforms identifiers to lowercase.
+func suppressPropertyTransformDiffs(
+	resourceToken string,
+	spec *metadata.CloudAPIResource,
+	diff *resource.ObjectDiff,
+	originalInputs resource.PropertyMap,
+) *resource.ObjectDiff {
+	transforms := GlobalTransformCache.GetOrCompile(resourceToken, spec)
+	if len(transforms) == 0 {
 		return diff
 	}
 
-	newFsp := updated.New.ObjectValue()
-	rop, hasRop := newFsp["replicationOverwriteProtection"]
-	if !hasRop || !rop.IsString() || rop.StringValue() != "REPLICATING" {
-		return diff
+	// Convert originalInputs to a map[string]interface{} for JSONata evaluation
+	inputsMap := propertyMapToMap(originalInputs)
+
+	// Process updates - these are changes to existing properties
+	for key, valueDiff := range diff.Updates {
+		path := string(key)
+		oldVal := resourcex.DecodeValue(valueDiff.Old)
+		newVal := resourcex.DecodeValue(valueDiff.New)
+
+		if shouldSuppressPropertyDiff(transforms, path, oldVal, newVal, inputsMap, spec.IrreversibleNames) {
+			glog.V(7).Infof("Suppressing diff for %s.%s via propertyTransform", resourceToken, path)
+			delete(diff.Updates, key)
+		}
+	}
+
+	// Process adds - new properties being added
+	for key, value := range diff.Adds {
+		path := string(key)
+		newVal := resourcex.DecodeValue(value)
+
+		// For adds, check if the transform would normalize the value to match
+		// what we expect (often nil/empty maps to a default)
+		if shouldSuppressPropertyAdd(transforms, path, newVal, inputsMap, spec.IrreversibleNames) {
+			glog.V(7).Infof("Suppressing add for %s.%s via propertyTransform", resourceToken, path)
+			delete(diff.Adds, key)
+		}
 	}
 
-	// AWS has set it to REPLICATING - suppress this change
-	delete(diff.Updates, fspKey)
 	return diff
 }
+
+// shouldSuppressPropertyDiff checks if a property diff should be suppressed
+// based on propertyTransform normalization.
+func shouldSuppressPropertyDiff(
```

Comment:
This currently re-diffs the full old/new values, but we already have a `ValueDiff` tree from Pulumi. I think we should instead drive the suppression from `ValueDiff.Object` / `ValueDiff.Array` so you only evaluate transforms for changed leaf paths, not all fields. e.g.

```go
// Instead of recursing via old/new maps/arrays, walk the diff tree:
//
// func shouldSuppressValueDiff(transforms []CompiledTransform, basePath string, diff resource.ValueDiff, ...)
//   - if diff.Object != nil: iterate diff.Object.Updates and recurse
//   - if diff.Array != nil: iterate diff.Array.Updates and recurse
//   - else: evaluateAndCompare for basePath using diff.Old/diff.New
```